app/ts/library/provider.ts

```diff
@@ -97,12 +97,17 @@ export const connectBrowserProvider = async (
 			provider.on('block', blockCallback)
 		}
 		accountsChangedCallback(await Promise.all(accounts.map(account => account.getAddress())))
-		if (block) blockCallback(block)
-	}
-	const blockCallback = (block: Block | null) => {
 		if (block) updateLatestBlock(block, store, blockInfo, signers)
 	}
 
+	const blockCallback = (blockNumber: number | null) => {
+		if (blockNumber) {
+			provider.getBlock(blockNumber).then((block) => {
+				if (block) updateLatestBlock(block, store, blockInfo, signers)
+			})
+		}
+	}
```

Comment:
A few different points of feedback all captured below:
1. Use async/await rather than `.then`
2. Handle errors, don't swallow them.
3. Use guard clauses rather than deeply nested conditionals.
```suggestion
	const blockCallback = async (blockNumber: number | null) => {
		try {
			if (!blockNumber) return
			const block = await provider.getBlock(blockNumber)
			if (!block) return
			updateLatestBlock(block, store, blockInfo, signers)
		} catch (error) {
			// TODO: handle error somehow
		}
	}
```

Reply:
For now there is no need to check or update any state if the getBlock or updateLastestBlock throws for some reason, so logging the error should be fine. If the same RPC that gave us the new block doesn't know about the new block or the network dies we can just wait on the next new block.